genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/GenieBaseTask.java

```diff
@@ -41,6 +42,13 @@
 @Slf4j
 public abstract class GenieBaseTask implements WorkflowTask {
 
+    protected static final String NO_ID_FOUND = "<no id>";
+    protected final Registry registry;
```

Comment:
I'd recommend using a getter rather than Protected here. If you extend a task twice you lose access to protected. Get is cleaner.